packages/hash/frontend/src/blocks/page/Comments/CommentBlock.tsx

```diff
@@ -130,13 +176,61 @@ export const CommentBlock: FunctionComponent<CommentProps> = ({ comment }) => {
       </Box>
 
       <Box p={0.5} pt={2} position="relative">
-        <CommentTextField
-          onLineCountChange={(lines) => setShouldCollapse(lines > 2)}
-          value={hasText}
-          className={collapsed ? styles.Comment__TextField_collapsed! : ""}
-          readOnly
-        />
-        {shouldCollapse && collapsed ? (
+        <Box
+          sx={({ palette, transitions }) => ({
+            border: `${editable ? 1 : 0}px solid ${palette.gray[30]}`,
+            pl: editable ? 2 : 0,
+            borderRadius: 1.5,
+            transition: transitions.create(["padding", "border-color"]),
+            "&:focus-within": {
+              borderColor: palette.blue[60],
+            },
+          })}
+        >
+          <CommentTextField
+            onLineCountChange={(lines) => setShouldCollapse(lines > 2)}
+            value={inputValue}
+            className={`${styles.Comment__TextField} ${
+              editable
+                ? styles.Comment__TextField_editable
+                : collapsed
+                ? styles.Comment__TextField_collapsed!
+                : ""
+            }`}
+            editable={editable}
+            readOnly={!editable}
+            placeholder="Edit comment"
+            onClose={resetCommentText}
+            onSubmit={handleEditComment}
+            onChange={setInputValue}
+          />
+        </Box>
+
+        <Collapse in={editable}>
+          <Box
+            sx={{
+              display: "flex",
+              gap: 0.75,
+              justifyContent: "flex-end",
+              pt: 0.75,
+            }}
+          >
+            <Button size="xs" variant="tertiary" onClick={resetCommentText}>
+              Cancel
+            </Button>
+            <Button
```

Comment:
Requested change: we should allow this button to accept different text to display in the loading indicator – when creating or editing a comment it should be 'Saving...' or similar rather than 'Loading...' as it is now

Reply:
Added this in 33386539c2520f9f4c6d3892276069b1749f52e3, changed the button loading text to "Saving...".